tests/core/daemon/test_daemon.py

```diff
@@ -57,8 +52,26 @@ async def get_daemon_with_temp_keyring(self, get_b_tools):
         async for _ in setup_daemon(btools=get_b_tools):
             yield get_b_tools
 
+    # TODO: Ideally, the db_version should be the (parameterized) db_version
+    # fixture, to test all versions of the database schema. This doesn't work
+    # because of a hack in shutting down the full node, which means you cannot run
+    # more than one simulations per process.
+    @pytest_asyncio.fixture(scope="function")
+    async def simulation(self, bt, get_b_tools, get_b_tools_1):
+        # get_b_tools._config["daemon_port"] = find_available_listen_port("daemon")
+        # save_config(get_b_tools.root_path, "config.yaml", get_b_tools._config)
+        async for _ in setup_full_system(
+            test_constants_modified,
+            bt,
+            b_tools=get_b_tools,
+            b_tools_1=get_b_tools_1,
+            connect_to_daemon_port=get_b_tools._config["daemon_port"],
+            db_version=1,
+        ):
+            yield _
+
     @pytest.mark.asyncio
-    async def test_daemon_simulation(self, simulation, get_daemon, get_b_tools, self_hostname):
+    async def test_daemon_simulation(self, simulation, bt, get_b_tools, get_b_tools_1):
```

Comment:
`test_daemon_simulation`: what is the goal of this test? What minimum configuration of nodes / processes is required? (Are we simulating 1 node running all services, and one node running ... ?)

Reply:
Not sure, I didn't write this one